components/ingest-service/integration_test/project_update_actions_test.go

```diff
@@ -0,0 +1,342 @@
+									Values: []string{"org1"},
+								},
+							},
+						},
+					},
+				},
+			},
+			projectIDs: []string{"project9"},
+		},
+		{
+			description: "chefServers: one rule two conditions with only one matching",
+			action: iBackend.InternalChefAction{
+				RemoteHostname:   "chef-server.org",
+				OrganizationName: "org1",
+				Projects:         []string{"old_tag"},
+			},
+			projects: map[string]*iam_v2.ProjectRules{
+				"project9": &iam_v2.ProjectRules{
+					Rules: []*iam_v2.ProjectRule{
+						&iam_v2.ProjectRule{
+							Conditions: []*iam_v2.Condition{
+								&iam_v2.Condition{
+									Type:   iam_v2.ProjectRuleConditionTypes_CHEF_SERVERS,
+									Values: []string{"chef-server2.org"},
+								},
+								&iam_v2.Condition{
+									Type:   iam_v2.ProjectRuleConditionTypes_CHEF_ORGS,
+									Values: []string{"org1"},
+								},
+							},
+						},
+					},
+				},
+			},
+			projectIDs: []string{},
+		},
+
+		// General
+		{
+			description: "project rules does not have any rules",
+			action: iBackend.InternalChefAction{
+				RemoteHostname:   "chef-server.org",
+				OrganizationName: "org1",
+				Projects:         []string{"old_tag"},
+			},
+			projects: map[string]*iam_v2.ProjectRules{
+				"old_tag": &iam_v2.ProjectRules{
+					Rules: []*iam_v2.ProjectRule{},
+				},
+			},
+			projectIDs: []string{},
+		},
+		{
+			description: "project rule does not have any conditions",
+			action: iBackend.InternalChefAction{
+				RemoteHostname:   "chef-server.org",
+				OrganizationName: "org1",
+				Projects:         []string{"old_tag"},
+			},
+			projects: map[string]*iam_v2.ProjectRules{
+				"old_tag": &iam_v2.ProjectRules{
+					Rules: []*iam_v2.ProjectRule{
+						&iam_v2.ProjectRule{
+							Conditions: []*iam_v2.Condition{},
+						},
+					},
+				},
+			},
+			projectIDs: []string{},
+		},
+	}
+
+	for _, test := range cases {
+		t.Run(fmt.Sprintf("action match: %s", test.description),
+			func(t *testing.T) {
+				test.action.RecordedAt = time.Now()
+				suite.IngestActions([]iBackend.InternalChefAction{test.action})
+				// Send a project rules update event
+				esJobID, err := suite.ingest.UpdateActionProjectTags(ctx, test.projects)
+				assert.Nil(t, err)
+
+				jobStatus, err := suite.ingest.JobStatus(ctx, esJobID)
+				assert.Nil(t, err)
+				for !jobStatus.Completed {
+					time.Sleep(time.Millisecond * 5)
+					jobStatus, err = suite.ingest.JobStatus(ctx, esJobID)
+					assert.Nil(t, err)
+					if err != nil {
+						assert.FailNow(t, "testing job complete")
+					}
+				}
+
+				suite.RefreshIndices(fmt.Sprintf("%s-%s", mappings.Actions.Index, "*"))
+
+				// assert the node's project IDs
+				actualActions, err := suite.GeActions(100)
+				assert.Nil(t, err)
+				assert.Equal(t, 1, len(actualActions), "wrong number of actions retrieved")
+
+				actualAction := actualActions[0]
+
+				assert.ElementsMatch(t, test.projectIDs, actualAction.Projects)
```

Comment:
I would definitely suggest using `require` here, because if an empty list is returned above you'll get a nil pointer exception trying to access the fields on this line.